Objects/longobject.c

```diff
@@ -4828,21 +4828,12 @@ long_divmod(PyObject *a, PyObject *b)
 static PyLongObject *
 long_invmod(PyLongObject *a, PyLongObject *n)
 {
-    PyLongObject *b, *c;
-
     /* Should only ever be called for positive n */
     assert(_PyLong_IsPositive(n));
 
-    b = (PyLongObject *)PyLong_FromLong(1L);
-    if (b == NULL) {
-        return NULL;
-    }
-    c = (PyLongObject *)PyLong_FromLong(0L);
-    if (c == NULL) {
-        Py_DECREF(b);
-        return NULL;
-    }
     Py_INCREF(a);
+    PyLongObject *b = (PyLongObject *)Py_NewRef(_PyLong_GetOne());
```

Comment:
@vstinner Out of curiosity: why did you add the `Py_NewRef` when `_PyLong_GetOne` is documented to return an immortal singleton?
(or can modern compilers optimize this away?)

Reply:
The Py_NewRef() is useless. I added it because the code wants a strong refererence and I didn't pay attention that these two constants are immortal.

>  (or can modern compilers optimize this away?)

No, Py_NewRef(<immortal>) is not optimized.

See also my PR gh-124076 attempt.